src/presentation/web/assignment_views.py

```
@@ -0,0 +1,363 @@
+            f"<td class='assignment-mark-weight'>{'-' if assignment.grade_type in ['S','U'] else ('%.2f' % float(assignment.mark_weight) if assignment.mark_weight is not None else '0.00')}</td>"
+            f"<td class='assignment-grade-type'>{assignment.grade_type}</td>"
+            f"<td class='flex gap-1'>"
+            f"<form method='post' action='/semester/{semester}/subject/{code}/assignment/{assessment}/{code}/{semester}/{year}/delete'>"
+            f"<input type='hidden' name='year' value='{year}' />"
+            f"<button class='btn btn-xs btn-error' type='submit'>✕</button>"
+            f"</form>"
+            f"<button class='btn btn-xs btn-outline' type='button' onclick=\"window.startInlineEditAssignment('{assessment}','{code}','{semester}','{year}')\">Edit</button>"
+            f"</td>"
+        )
+        return JSONResponse({"success": True, "row_html": row_html})
+    except Exception as e:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)